src/components/inputs/input-account/near-amount-account/NearAmountAccount.tsx

```diff
@@ -0,0 +1,352 @@
+            valid: false,
+          },
+          editable: true,
+          cleared: false,
+        },
+      }
+    }
+
+    return auxState
+  }
+
+  const handleAddNewItem = () => {
+    const id = uuid()
+
+    setState({
+      ...state,
+      [id]: {
+        account: {
+          value: '',
+          valid: false,
+        },
+        amount: {
+          value: 0,
+          valid: false,
+        },
+        editable: true,
+        cleared: false,
+      },
+    })
+  }
+
+  const handleRemoveItem = (id: string) => {
+    setUsed(used - state[id].amount.value)
+    const newState = { ...state }
+
+    newState[id].amount.value = 0
+    newState[id].account.value = ''
+    newState[id].amount.valid = false
+    newState[id].account.valid = false
+    newState[id].cleared = true
+    setState(newState)
+    ;(document.getElementById(`amount-${id}`) as HTMLInputElement).value = ''
+    ;(document.getElementById(`account-${id}`) as HTMLInputElement).value = ''
+  }
+  const handleChangeAmount = (id: string, amount: number) => {
+    const newState = { ...state }
+    newState[id].amount.value = amount
+    newState[id].cleared = false
+    setState(newState)
+    if (newState[id].amount.valid) {
+      setUsed(
+        Object.values(newState)
+          .filter((elm) => elm.amount.valid)
+          .reduce((prev, curr) => {
+            if (curr.amount.valid) {
+              return curr.amount.value + prev
+            }
+            return 0
+          }, 0)
+      )
+    }
+  }
+
+  const handleChangeAccount = async (id: string, account: string) => {
+    const newState = { ...state }
+    newState[id].account.value = account
+    newState[id].cleared = false
+    setState(newState)
+  }
+
+  const validateAmountById = (id: string, valid: boolean) => {
+    if (valid) setUsed(used + state[id].amount.value)
+    const newState = { ...state }
+    newState[id].amount.valid = valid
+    setState(newState)
+  }
+
+  const validateAccountById = (id: string, valid: boolean) => {
+    const newState = { ...state }
+    newState[id].account.valid = valid
+    setState(newState)
+  }
+
+  const validateAmount = (id: string, amount: number) => {
+    const newState = { ...state }
+    delete newState[id]
+
+    const valid = transferTemplate
+      ? transferTemplate.available - used - amount >= 0 && amount > 0
+      : totalAmount - used - amount >= 0 && amount > 0
+
+    validateAmountById(id, valid)
+
+    return valid
+  }
+
+  const validateAccount = async (id: string, account: string) => {
+    const valid = await accountExists(account)
+
+    validateAccountById(id, valid)
+
+    return valid
+  }
+
+  const reset = () => {
+    setState({
+      ...defaultState,
+      ...addFieldsToState(0),
+    })
+    setUsed(0)
+    Object.keys(state).forEach((id) => {
+      const amountInput = document.getElementById(
+        `amount-${id}`
+      ) as HTMLInputElement
+      const accountInput = document.getElementById(
```

Comment:
thing is like if we populate the value of the input with state, it is not possible to change it.
i'm cleaning the DOM as well when the user selects the trash icon because it's the new behaviour, instead of just remove the entire line